domain/state.go

```diff
@@ -95,3 +97,65 @@ func (st *StateBase) Prepare(query string, typeSamples ...any) (*sqlair.Statemen
 	st.statements[query] = stmt
 	return stmt, nil
 }
+
+type processFunc = func(in, out, samples []any) ([]any, []any, []any)
+
+func In[T any](v T) processFunc {
+	return func(in, out, samples []any) ([]any, []any, []any) {
+		return append(in, v), out, append(samples, v)
+	}
+}
+
+func Out[T any](v *T) processFunc {
+	return func(in, out, samples []any) ([]any, []any, []any) {
+		return in, append(out, v), append(samples, *v)
+	}
+}
+
+func MultiOut[T any](v *[]T) processFunc {
```

Comment:
Is there a better name? I realise `InOut` is not strictly correct, but is there something we can live with that is more self explanatory from a user reading it for the first time. In and Out seem intuitive so that's good for those.

Reply:
`Out1`, `Outm`?